var/spack/repos/builtin/packages/gtkplus/package.py

```diff
@@ -29,10 +29,13 @@ class Gtkplus(Package):
     """The GTK+ 2 package contains libraries used for creating graphical user
        interfaces for applications."""
     homepage = "http://www.gtk.org"
+    url = "http://ftp.gnome.org/pub/gnome/sources/gtk+/2.24/gtk+-2.24.31.tar.xz"
 
     version(
-        '2.24.25', '612350704dd3aacb95355a4981930c6f',
-        url="http://ftp.gnome.org/pub/gnome/sources/gtk+/2.24/gtk+-2.24.25.tar.xz")
+        '2.24.31',
+        '68c1922732c7efc08df4656a5366dcc3afdc8791513400dac276009b40954658')
+    version(
+        '2.24.25', '612350704dd3aacb95355a4981930c6f')
```

Comment:
The published hash is probably better than the ones we used. Can you add the published hash for the previous versions too? Btw, we have several hacks for flake8 that only get run when you do `spack flake8`, so if you are trying to use your own flake8 it might behave differently.